run.sh

```diff
@@ -0,0 +1,171 @@
+#!/bin/sh
+set -e
+PROGNAME=$(basename $0)
+COMPONENTS_DIR=/home/system/components
+die() {
+    echo "$PROGNAME: $*" >&2
+    exit 1
+}
+
+usage() {
+    if [ "$*" != "" ] ; then
+        echo "Error: $*"
+    fi
+    cat << EOF
+
+Usage: $PROGNAME [OPTION ...] [foo] [bar]
+
+-----------------------------------------------------------------
+|  CLI tool for building and managing Edgebox updates           |
+-----------------------------------------------------------------
+
+Options:
+-h, --help               display this usage message and exit
+-c, --check              check for updates
+-u, --update             update the system
+
+EOF
+    exit 1
+}
+
+# Update check. This function is called when the --check flag is passed
+# It should check for updates by doing the following:
+# 1. Open the file versions.env and read the current version for each component (ws, api, apps, logger, edgeboxctl)
+# 2. Check in each component folder for the available git tags
+# 3. If there is a newer (semver) tag than the one read from the file, save it to a variable
+# 4. Write the new available next version to the versions.env file
+check() {
+    echo "Checking for updates..."
+
+    # Remove the file targets.env if it exists
+    if [ -f $COMPONENTS_DIR/updater/targets.env ]; then
+        rm $COMPONENTS_DIR/updater/targets.env
+    fi
+
+    # Check for updates for each component
+    for component in ws api apps logger edgeboxctl; do
+        
+        # Convert to uppercase
+        component_upper=$(echo "$component" | tr '[:lower:]' '[:upper:]')
+
+        # echo "Checking for updates for $component_upper..."
+        cd $COMPONENTS_DIR/$component
+
+        # Check if versions.env file exists
+        if [ ! -f $COMPONENTS_DIR/updater/versions.env ]; then
```

Comment:
how does this work for apps?  since they are in a monorepo?

Reply:
It will work just the same (for now).
As you also contributed to the [apps](https://github.com/edgebox-iot/apps) repo, you've seen that we now want to build versioned releases of our app docker images.

We should also create a gh release each time we update or add a new app. Although this is not the case right now, the idea is that we support a stable and fully working set of apps, and each update to the apps repo releases is always going to be for a stable app addition or update. 

This has the disadvantage of not allowing the user to control independently what edgeapp versions they run, but I believe we can deal with this case independently sometime later.